packages/react/src/index.test.tsx

```diff
@@ -72,7 +72,8 @@ describe('HoneybadgerReact', () => {
       sandbox.spy(honeybadger, 'notify')
       sandbox.spy(honeybadger, 'showUserFeedbackForm')
 
-      const MyError = jest.fn(() => 'custom error view')
+      // need to cast to unknown because ErrorComponent does not expect a jest.Mock type
```

Comment:
I'm not happy with this, but I spend too much time trying to get it right. I don't know why I can't get it work 🤷

Reply:
Ah yeah, the jest mock type is getting in the way! Can you try something like:

```
const ErrComponent = () => <>custom error view</>
const MockErrComponent = ErrComponent as jest.MockedFunction<typeof ErrComponent>
      
TestRenderer.create(<HoneybadgerErrorBoundary honeybadger={honeybadger} showUserFeedbackFormOnError={true} ErrorComponent={MockErrComponent}><Broken /></HoneybadgerErrorBoundary>)
```

This is helpful: https://www.benmvp.com/blog/using-jest-mock-functions-typescript/

Reply:
Thanks! I tried it and it didn't work.

And actually, in the snippet above, there is no call to jest.mock or jest.fn so the ErrComponent is never mocked.
In the example you showed, there component was a separate file, and was mocked with `jest.mock('./ComponentFile')` before being used.

I already tried this and faced other issues:
```
// added these at the top of the tests file
import DefaultErrorComponent from './DefaultErrorComponent';

jest.mock('./DefaultErrorComponent');

// then, I modified the test to simply use DefaultErrorComponent
TestRenderer.create(<HoneybadgerErrorBoundary honeybadger={honeybadger} ErrorComponent={DefaultErrorComponent}><Broken /></HoneybadgerErrorBoundary>)
```

With the above code, the test is now failing with this:
```
expect(jest.fn()).toBeCalledWith(...expected)

    - Expected
    + Received

      Object {
    -   "error": Any<Error>,
    -   "errorOccurred": Any<Boolean>,
    -   "info": Object {
    -     "componentStack": Any<String>,
    -   },
    +   "error": [Error: Busted, sorry],
    +   "errorOccurred": true,
    +   "info": null,
      },
      {},

    Number of calls: 1
```

The issue here is that `info` is null and after diving into it, it seems to be because React will call render more than once (which is fine), and will eventually call it with all the necessary props. I will take another stab at it with `jest.mock('./DefaultErrorComponent')`. Perhaps we should change the way we make our assertions.

Reply:
Huh weird, I tested this before recommending it to you... let me try again.

Reply:
Ah ok mine does seem to pass, but I made a typo when I posted it to you. My apologies! Can you give this a shot? 

```
it('should render a custom error message when a component errors', (done) => {
      sandbox.spy(honeybadger, 'notify')
      sandbox.spy(honeybadger, 'showUserFeedbackForm')

      const MyErrComponent = () => <>custom error</>
      const MyErrComponentMock = jest.fn(MyErrComponent) as jest.MockedFunction<typeof MyErrComponent>
      
      TestRenderer.create(<HoneybadgerErrorBoundary honeybadger={honeybadger} ErrorComponent={MyErrComponentMock}><Broken /></HoneybadgerErrorBoundary>)
      expect(MyErrComponentMock).toBeCalledWith({
        error: expect.any(Error),
        info: { componentStack: expect.any(String) },
        errorOccurred: expect.any(Boolean)
      }, {})
      // Still want to ensure notify is only called once. The MyErrComponent component will be created twice by React.
      afterNotify(done, function () {
        assert.calledOnce(honeybadger.notify as SinonSpy)
        assert.notCalled(honeybadger.showUserFeedbackForm as SinonSpy)
      })
    })
```

Reply:
It works, huuuuuge thank you! 🔥